File: src/leetcode/hash_maps_and_sets/1207_unique_number_of_occurrences.ts
const uniqueOccurrences = (arr: number[]): boolean => {
  const hashMap = arr.reduce((acc, item) => acc.set(item, (acc.get(item) ?? 0) + 1), new Map<number, number>());

  return hashMap.size === new Set(hashMap.values()).size;
};

export default uniqueOccurrences;

File: src/leetcode/hash_maps_and_sets/1207_unique_number_of_occurrences.rb
# frozen_string_literal: true

# LeetCode 75
# https://leetcode.com/problems/unique-number-of-occurrences/description/?envType=study-plan-v2&envId=leetcode-75

# Time complexity: O()
# Space complexity: O()

# @param {Integer[]} arr
# @return {Boolean}
def unique_occurrences(arr)
  hash_map = arr.each_with_object(Hash.new(0)) { |item, acc| acc[item] += 1 }

  # in-built
  # hash_map = arr.tally

  hash_map.length == hash_map.values.uniq.length
end
